File: 0724-find-pivot-index/0724-find-pivot-index.cpp
class Solution {
public:
    int pivotIndex(vector<int>& nums) {
        int n = nums.size();
        vector<int> sums(n + 1);
        int sum = 0;

        for (int i = 0; i < n; i++) {
            sum += nums[i];
            sums[i + 1] = sum;
        }

        for (int i = 0; i < n; i++) {
            if (sums[i] == sums[n] - sums[i + 1]) {
                return i;
            }
        }

        return -1;
    }
};

File: 0724-find-pivot-index/0724-find-pivot-index.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        n = len(nums)
        prefix = [nums[0]]
        
        for i in range(1, n):
            prefix.append(prefix[i-1] + nums[i])
            
        for i in range(n):
            if (i == 0):
                l = 0
            else:
                l = prefix[i-1]
                
            if (i == n - 1):
                r = 0
            else:
                r = prefix[-1] - prefix[i+1] + nums[i+1]
            
            if (l == r):
                return i
            
        return -1